packages/client/src/context/FetchCategoriesAndMovies.jsx

```diff
@@ -0,0 +1,46 @@
+import React, {
+  createContext,
+  useContext,
+  useReducer,
+  useEffect,
+  useMemo,
+} from 'react';
+import {
+  ReducerForFetchMoviesAndCategories,
+  initialState,
+  handleCategoriesFetch,
+  handleMovieFetch,
+} from '../containers/MovieListPage/ReducerForFetchMoviesAndCategories';
+
+const MoviesAndCategoriesContext = createContext();
+
+export function MoviesAndCategoriesContextProvider({ children }) {
+  const [state, dispatch] = useReducer(
```

Comment:
Once again, should be separated if we even need both, and the naming of the context should be
SomethingContext. 
Saying fetchSomethingSomething makes it sound like a helper or a hook and it is not intuitive for an outsider 🙏🏽